app/concepts/todo/create.rb

```diff
@@ -1,50 +1,30 @@
 class Todo::Create < Trailblazer::Operation
+  include Model
+  model Todo, :create
 
-  # Logic for creating a TODO
-  #     if `todo_list` is specified
-  #         create the `todo`
-  #         associated with `todo_list`
-  #     else
-  #         create the `todo`
-  #         create a `user`
-  #         create a default `todo_list`
-  #         associate `todo_list` with `user`
-  #         associate `todo` with `todo_list`
-  def process(params)
-    # make sure title is not blank
-    return invalid! if params[:todo][:title].blank?
+  contract do
+    property :title
+    property :todo_list_id
+
+    validates :title, presence: true
+  end
 
-    # build the todo
-    todo = @model = Todo.new(params.require(:todo).permit(:title, :description))
+  def process(params)
+    validate(params[:todo]) do |f|
+      f.todo_list_id = associate_list(params[:todo_list_id])
+      f.save
+    end
+  end
 
-    # find or create the parent list for the todo
-    if params[:todo_list_id].blank?
-      todo_list = TodoList.find_or_create_by(name: "Default To-do List")
+  private
 
-      # Find or create the owner of the todo_list:
-      ##### If current_user_id is present, fetch that user
-      ##### otherwise, create a new one
-      current_user =
-        if params[:current_user] == nil
-          if params[:current_user_id] == nil
-            User.create!(fullname: "Guest")
-          else
-            User.find(params[:current_user_id])
-          end
-        else
-          params[:current_user]
-        end
+  def associate_list(todo_list_id)
+    return todo_list_id if todo_list_id.present?
 
-      # assign the todo_list to the user
-      todo_list.user = current_user
-      todo_list.save!
-    else # i.e. if params[:todo_list_id] is not nil
-      todo_list = TodoList.find(params[:todo_list_id])
-    end
+    default_todo = TodoList.where(name: 'Default To-do List').first
+    return default_todo.id if default_todo.present?
 
-    # associate the todo with the todo_list
-    todo.list = todo_list
-    todo.save!
+    todo_list_attr = { name: 'Default To-do List' }
+    TodoList::Create.(todo_list: todo_list_attr).model.id
```

Comment:
I just didnt want to put an if-else logic inside the ```process``` method.  
Can you suggest a better location for this logic?